### DISCUSSION TITLE
Forcing an album into a compilation

### DISCUSSION THREAD

**Discussion by hanrahs** [rdswift/picard-scripts#7] [Q&A] [answered]
Hi, 
First of all, thank you very much for sharing your naming script. I have been able to adapt it to how I would like my files named and I'm not sure I would have been able to do that with out the base you provided.

I have a set of compilations that all have Various Artists, except for one which has two artists listed (the compilation is called Zydeco Swamp Vol 7). I would like to keep all of these together in my compilations folder (Various Artists folder in your original code) so I was thinking the best way would be a tagger script, I feel this is a fairly rare case - not often you would have a series of compilations where most are tagged Various Artists and one isn't) 

given this code on rows 422-423 of your original code

       $if($eq(%_nAlbumType%,Standard),$if($eq($if2(%musicbrainz_albumartistid%,%_cVariousArtistID%),%_cVariousArtistID%),
             $set(_nFilePath,%_cVariousArtist%/%_nYear% %_nANT%/),

and this on row 115

       $set(_cVariousArtistID,89ad4ac3-39f7-470e-963a-56509c546377)

my read was that if I made the musicbrainz_albumartistid = cVariousArtistID I should have it sorted.

so I made this tagger script

      $set(%musicbrainz_albumartistid%,89ad4ac3-39f7-470e-963a-56509c546377)

but that does not work, nor does changing the tags manually in the metadata window of Picard. It just keeps rolling through the if statements and is treated like a standard album in the end.

I'm not really sure where to go next, this has stumped me. Would appreciate any help or ideas I can try. I feel like it is a really obvious answer that I am just missing. I guess I could just write a simple naming script for this particular case, but I'm annoyed I can't work it out now.

Thanks

**Comment by hanrahs** [comment]
I initially tried removing the  musicbrainz_albumartistid tag in Picards metadata window so it would use %_cVariousArtistID% here:

        $if2(%musicbrainz_albumartistid%,%_cVariousArtistID%) 

but that did not work which is why I tried changing it.

**Comment by rdswift** [comment, marked as answer]
Try changing Line 422 to the following and see if that helps:

```taggerscript
$if($eq(%_nAlbumType%,Standard),$if($or($in(%musicbrainz_albumartistid%,;),$eq($if2(%musicbrainz_albumartistid%,%_cVariousArtistID%),%_cVariousArtistID%)),
```

**Reply by hanrahs** [reply]
Thank you very much, that is working for me.

What is actually happening here? I'm not sure I have come across the ";" used like this

       $in(%musicbrainz_albumartistid%,;)

**Reply by rdswift** [reply]
If there is more than one album artist listed, the `%musicbrainz_albumartistid%` will list the MBIDs for each artist separated by semicolons.  All this does is check if there is a semicolon in the tag, which indicates multiple album artists.  This isn't foolproof because it could result in false positives, but that should be a rare occurrence.

**Reply by hanrahs** [reply]
I need to think of the code as simple snippets that read and save text rather than a programming language, I think that might help.

The fix has had quite a bit more effect moving unintended albums to compilations so I added that is also needs to have a secondary release type as compilation - This seems to have fixed it for now, I am sure I will find some exceptions  - but it should make it easier to manipulate one way or the other with a tagger script if needed.

Once again thanks for the help
